doc/source/whatsnew/v0.20.0.txt

```diff
@@ -38,8 +38,127 @@ Other enhancements
 Backwards incompatible API changes
 ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
 
-.. _whatsnew_0200.api:
 
+Map on Index types now return other Index types
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+-  ``map`` on an ``Index`` now returns an ``Index``, not an array (:issue:`12766`)
+
+  .. ipython:: python
+
+     idx = Index([1, 2])
+     idx
+
+  Previous Behavior:
+
+  .. code-block:: ipython
+
+     In [3]: idx.map(lambda x: x * 2)
+     Out[3]: array([2, 4])
+
+  New Behavior:
+
+  .. ipython:: python
+
+      idx.map(lambda x: x * 2)
+
+-  ``map`` on an ``Index`` or ``MultiIndex`` returns the appropriate type depending on output dimensionality
+
+  .. ipython:: python
+
+     mi = MultiIndex.from_tuples([(1, 2), (2, 4)])
```

Comment:
put this with where you define the Index as well. You can then put the previous/new for Multi with Index